routes/foodtrucks.go

```diff
@@ -149,3 +167,117 @@ func GetFoodTrucksHandler(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 	json.NewEncoder(w).Encode(foodTrucks)
 }
+
+func PutFoodTrucksHandler(w http.ResponseWriter, r *http.Request) {
+
+	// Checks for food truck ID
+	params := mux.Vars(r)
+	foodTruckID, foodTruckIDExists := params["foodTruckID"]
+	if !foodTruckIDExists {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// Get user from context
+	user, userLoggedIn := r.Context().Value(middleware.UserKey).(string)
+
+	fmt.Println(user)
+
+	// Check for a user, or if the user agent is from the scraper
+	if !userLoggedIn && r.Header.Get("User-Agent") != "MunchCritic/1.0" {
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	foodTruckDecoder := json.NewDecoder(r.Body)
+	foodTruckDecoder.DisallowUnknownFields()
+
+	// Decode request
+	var currentFoodTruck updateFoodTruckRequest
+	err := foodTruckDecoder.Decode(&currentFoodTruck)
+	if err != nil {
+		log.Printf("ERROR: %v", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// Set tags to an empty array if they don't exist
+	tags := currentFoodTruck.Tags
+	if tags == nil {
+		tags = []string{}
+	}
+
+	// Determine which fields should be updated
+	var updateData bson.D
+
+	if currentFoodTruck.Name != "" {
+		updateData = append(updateData, bson.E{"name", currentFoodTruck.Name})
+	}
+	if currentFoodTruck.Address != "" {
+		updateData = append(updateData, bson.E{"address", currentFoodTruck.Address})
+	}
+	if len(currentFoodTruck.Location) != 0 {
+		updateData = append(updateData, bson.E{"location", currentFoodTruck.Location})
+	}
+	if currentFoodTruck.Owner != "" {
+		updateData = append(updateData, bson.E{"owner", currentFoodTruck.Owner})
+	}
+	if currentFoodTruck.Status != false {
+		fmt.Println("changed status")
+		updateData = append(updateData, bson.E{"status", currentFoodTruck.Status})
+	}
+	if currentFoodTruck.AvgRating != 0 {
+		updateData = append(updateData, bson.E{"avgRating", currentFoodTruck.AvgRating})
+	}
+	// Validate hours if updating
+	if currentFoodTruck.Hours[0][0] != "" {
+		fmt.Println("---going into time")
+		for i := 0; i < 7; i++ {
+			validOpenTime, err := regexp.MatchString(`^\d{2}:\d{2}$`, currentFoodTruck.Hours[i][0])
+			if err != nil {
+				log.Printf("ERROR: %v", err)
+				w.WriteHeader(http.StatusInternalServerError)
```

Comment:
StatusBadRequest is more fitting, since the error is on the user's part and not the server